rd_server.py

```diff
@@ -338,21 +341,25 @@ def scheduler_tick():
             slot.clear_work()
             free_slots.append(slot)
     # fill empty slots with new work
-    if len(work_list) != 0:
-        if len(free_slots) != 0:
-            slot = free_slots.pop()
-            work = work_list[0]
-            # search for image work if there is only one slot available
-            # allows prioritizing image runs without making scheduler the bottleneck
-            if len(free_slots) == 0:
-                try:
-                    work = find_image_work(work_list, work)
-                except Exception as e:
-                    rd_print(None, e)
-                    rd_print(None, 'Finding image work failed.')
-            work_list.remove(work)
-            rd_print(work.log,'Encoding',work.get_name(),'on',slot.machine.host)
-            slot.start_work(work)
+    for arch in arches:
+        free_slots_filtered = [slot for slot in free_slots if slot.arch == arch]
+        work_list_filtered = [work for work in work_list if work.arch == arch]
+        if len(work_list_filtered) != 0:
+            if len(free_slots_filtered) != 0:
```

Comment:
nit: [PEP-8 says](https://docs.python.org/3/library/stdtypes.html#truth-value-testing) you should just do:
```
if work_list_filtered:
   if free_slots_filtered:
```